platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -1555,6 +1555,10 @@ DisplayServerX11::WindowID DisplayServerX11::get_window_at_screen_position(const
 	return found_window;
 }
 
+ObjectID DisplayServerX11::get_focused_window_or_popup() const {
+	return window_get_attached_instance_id(_get_focused_window_or_popup());
```

Comment:
When there's no popups, `_get_focused_window_or_popup` is using the last focused window, not the current focused window, so it won't work if non of the windows is focused. But this probably should be fixed in the `_get_focused_window_or_popup` implementations.

Reply:
Thank you for this information. I didn't test this case.
I have updated the PR and reset the variable to `INVALID_WINDOW_ID`, when a window loses focus (on Linux, Windows and macOS)

Since this is a change in behavior, I would consider this PR now as experimental, especially because I can't test the implementation on macOS.